multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/cf/util/DeployedAfterModulesContentValidator.java

```diff
@@ -38,15 +50,26 @@ public void validate(List<Module> modules) {
         }
     }
 
-    private boolean areModuleDependenciesResolvable(Module module, Set<String> modulesInModelNames) {
+    private boolean areModuleDependenciesResolvable(Module module, Set<String> moduleNamesForDeployment) {
         List<String> moduleDependencies = module.getDeployedAfter();
         if (moduleDependencies == null) {
             return true;
         }
         return moduleDependencies.stream()
-                                 .filter(dependency -> !modulesInModelNames.contains(dependency))
-                                 .map(dependencyModule -> client.getApplication(dependencyModule, false))
-                                 .noneMatch(Objects::isNull);
+                                 .allMatch(dependency -> isModuleDependencyResolvable(moduleNamesForDeployment, module.getName(), dependency));
+    }
+
+    private boolean isModuleDependencyResolvable(Set<String> moduleNamesForDeployment, String moduleWithDependency, String dependencyModule) {
```

Comment:
Wouldn't 'moduleNamesForDeployment' contain non-application modules as well?

Reply:
yes, there is a check if the module is app several lines below